FluentTerminal.App.ViewModels/Profiles/ProfileProviderViewModelBase.cs

```diff
@@ -0,0 +1,403 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web;
+using FluentTerminal.App.Services;
+using FluentTerminal.Models;
+using FluentTerminal.Models.Enums;
+using GalaSoft.MvvmLight;
+
+namespace FluentTerminal.App.ViewModels.Profiles
+{
+    /// <summary>
+    /// Base class for all profile view model classes. It contains properties shared by all other view models
+    /// (theme-related properties, line-ending translation, and WinPTY/ConPTY selection).
+    /// </summary>
+    public abstract class ProfileProviderViewModelBase : ViewModelBase
+    {
+        #region Static
+
+        private static readonly LineEndingStyle[] LineEndingStylesArray =
+            Enum.GetValues(typeof(LineEndingStyle)).Cast<LineEndingStyle>().ToArray();
+
+        #endregion Static
+
+        #region Fields
+
+        protected readonly ISettingsService SettingsService;
+        protected readonly IApplicationView ApplicationView;
+
+        #endregion Fields
+
+        #region Properties
+
+        private ShellProfile _model;
+
+        public ShellProfile Model
+        {
+            get => _model;
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException();
+                }
+
+                if (ReferenceEquals(_model, value))
+                {
+                    return;
+                }
+
+                if (_model != null && _model.GetType() != value.GetType())
+                {
+                    throw new ArgumentException($"The value has to be of type {_model.GetType().Name}.");
+                }
+
+                _model = value;
+
+                LoadFromProfile(value);
+            }
+        }
+        public ObservableCollection<TabTheme> TabThemes { get; }
+
+        public ObservableCollection<TerminalTheme> TerminalThemes { get; }
+
+        public ObservableCollection<LineEndingStyle> LineEndingStyles { get; } =
+            new ObservableCollection<LineEndingStyle>(LineEndingStylesArray);
+
+        private LineEndingStyle _lineEndingTranslation;
+
+        public LineEndingStyle LineEndingTranslation
+        {
+            get => _lineEndingTranslation;
+            set => Set(ref _lineEndingTranslation, value);
+        }
+
+        private TabTheme _selectedTabTheme;
+
+        public TabTheme SelectedTabTheme
+        {
+            get => _selectedTabTheme ?? (_selectedTabTheme = TabThemes.FirstOrDefault(t => t.Id.Equals(_tabThemeId)));
+            set
+            {
+                TabTheme theme = value;
+
+                if (theme == null)
+                {
+                    // How to handle attempt to setting null theme?
+                    // 1) Throw an exception:
+                    //throw new ArgumentNullException();
+                    // 2) Ignore the attempt:
```

Comment:
I think 2) would be better, since an attempt to set it null will most probably be some binding doing stupid things

Reply:
Changed to ignore.